resources/assets/js/annotations/components/annotationCanvas.vue

```diff
@@ -512,6 +513,21 @@ export default {
         handleNewFeature(e) {
             if (this.hasSelectedLabel) {
                 let geometry = e.feature.getGeometry();
+                if (geometry.getType() === 'Polygon') {
+                    let polygonValidator = new PolygonValidator(e.feature);
+                    if (polygonValidator.isInvalidPolygon()) {
+                        this.$emit('is-invalid-polygon');
+                        this.annotationSource.once('change', () => {
```

Comment:
Why do this in an event handler instead of immediately?

Reply:
The 'requires-selected-label' issue is handled the same way, where an event is fired. So I just copied that.
But it's no problem to change that.

Reply:
I see, so there is probably more into it. But I don't remember why I did it that way :smiley: If in doubt and it works like that, just leave it.